lib/administrate/field/has_many.rb

```diff
@@ -25,7 +25,9 @@ def associated_resource_options
       end
 
       def selected_options
-        data && data.map { |object| object.send(primary_key) }
+        return if data.empty?
```

Comment:
None that I can see or think of...

The only place `selected_options` is used is as the 2nd argument in a call to [`options_for_select`](https://github.com/thoughtbot/administrate/blob/master/app/views/fields/has_many/_form.html.erb#L27). Tracing that method, when the 2nd argument is `nil` it seems to get turned into an empty array anyway (see [here](https://github.com/rails/rails/blob/66a7cfa91045e05f134efc9ac0e226e66161e2e6/actionview/lib/action_view/helpers/form_options_helper.rb#L354) and [here](https://github.com/rails/rails/blob/66a7cfa91045e05f134efc9ac0e226e66161e2e6/actionview/lib/action_view/helpers/form_options_helper.rb#L781)), however I'm wondering if it's just better to leave the previous behavior and not introduce any potential breaking changes. 

I don't think there's any downside to leaving it as `nil` except for the presence of the guard clause there.